### PR TITLE
fix: avoid importing @babel/polyfill  | closes #123

### LINKED ISSUE #123: [metal-debounce] Stop including @babel/polyfill
[Importing a polyfill](https://github.com/metal/metal-plugins/blob/adbdba263708563f325da806e6ea36681f999884/packages/metal-debounce/src/debounce.js#L3) directly doesn't seem like a good solution, especially when we can implement this function without the spread operator. (And in anycase the transpiler should take care of this not us)

The current version looks like this:

```javascript
function debounce(fn, delay) {
	return function debounced() {
		let args = arguments;
		cancelDebounce(debounced);
		debounced.id = setTimeout(function() {
				fn(...(null, args));
				}, delay);
	};
}
```

And it can be re-written like this:

```javascript
function debounce(fn, delay) {
	return function debounced() {
		let args = Array.prototype.slice.call(arguments, 0);
		cancelDebounce(debounced);
		debounced.id = setTimeout(function() {
			fn.apply(null, args);
		}, delay);
	};
}
```

*NOTE*: I'm aware that using `apply` is not considered a good practise when using ES6+ by some, but neither is using `arguments` so at this point does it really matter?

That being said, once the format task kicks in (`pretter-eslint`) the snippet above gets transformed (before it gets transpiled) to:

```
function debounce(fn, delay) {
	return function debounced() {
		let args = Array.prototype.slice.call(arguments, 0);
		cancelDebounce(debounced);
		debounced.id = setTimeout(function() {
			fn(...args);       // 👈no more apply, spread is here again
		}, delay);
	};
}
```

Which gets transpiled to:

```javascript
function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }  // 👈Array.from is expected to be available, which is not the case in IE so how do we solve this?

function debounce(fn, delay) {
  return function debounced() {
    var args = Array.prototype.slice.call(arguments, 0);
    cancelDebounce(debounced);
    debounced.id = setTimeout(function () {
      fn.apply(undefined, _toConsumableArray(args));
    }, delay); 
  };
}
```
 
This is probably because our `.eslintrc` extends `eslint-config-liferay` which extends `eslint-config-google` which [has](https://github.com/google/eslint-config-google/blob/master/index.js#L344-L345) 

```json
'prefer-rest-params': 2, 
'prefer-spread': 2
```

Which does actually [change](https://eslint.org/docs/rules/prefer-spread) `apply` for spread.

If any one has an idea on how to solve this without changing our config or massive changes, I'd appreciate your feedback.

Thanks

### PR DESCRIPTION
**Pull request by julien**

### CODE REVIEW

**Comment by wincent** [comment]
I don't think we want some of those formatting changes (eg. single quotes to doubles).

You probably want to add a `.prettierrc.json` containing the "standard":

```json
{
	"trailingComma": "es5",
	"bracketSpacing": false,
	"jsxBracketSameLine": true,
	"singleQuote": true,
	"tabWidth": 4,
	"useTabs": true
}
```

**Comment by jbalsas** [comment]
Looks like we might have an outdated browser saucelabs configuration:

```
24 05 2019 17:00:03.924:ERROR [launcher.sauce]: Can not start android 4.4 (Linux)
  [init({"version":"4.4","platform":"Linux","tags":[],"name":"metal-plugins tests","tunnel-identifier":"324.3","record-video":false,"record-screenshots":false,"device-orientation":null,"disable-popup-handler":true,"build":"324","public":null,"customData":{},"base":"SauceLabs","browserName":"android"})] The environment you requested was unavailable.
  Misconfigured -- Unsupported OS/browser/version/device combo: OS: 'unspecified', Browser: 'android', Version: '4.4.', Device: 'Android Emulator'
```

Other than that, LGTM.

File: packages/metal-debounce/src/debounce.js [hunk 1]
```diff
@@ -12,10 +8,10 @@ import from from 'core-js/es6/array'; //eslint-disable-line
  */
 function debounce(fn, delay) {
 	return function debounced() {
-		let args = arguments;
+		let args = Array.prototype.slice.call(arguments, 0);
```
Comment (wincent) [inline packages/metal-debounce/src/debounce.js]: Strictly speaking, you don't need the `0` here.

**Comment by julien** [comment]
I think we can safely close this one due to inactivity

**Pull request closed without merging**